src/Makefile

```diff
@@ -34,8 +34,8 @@ NVCC_GENCODE ?= -gencode=arch=compute_35,code=sm_35 \
                 -gencode=arch=compute_70,code=compute_70
 endif
 
-NVCUFLAGS  := -ccbin $(CXX) $(NVCC_GENCODE) -std=c++11
-CXXFLAGS   := -std=c++11
+NVCUFLAGS  := -ccbin $(CXX) $(NVCC_GENCODE) -std=c++17
+CXXFLAGS   := -std=c++17
```

Comment:
I don't think we can force all users to move to c++17 just for this feature.

Reply:
I agreed. I can feature-ize the compiling to C++17 only for GLOO.